web/src/engine/websites/MangaLib.ts

```diff
@@ -1,46 +1,125 @@
     }
 
     public override get Icon() {
         return icon;
     }
 
+    public override async Initialize(): Promise<void> {
+        const { data: { imageServers } } = await FetchJSON<APIResult<ImageServers>>(new Request(new URL('constants?fields[]=imageServers', this.apiUrl)));
+        this.imageServer = imageServers.find(server => server.id === 'main' && server.site_ids.includes(this.siteId));
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/ru/manga/[^/]`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaSlug = new URL(url).pathname.split('/').at(-1);
+        const { data: { slug_url, rus_name } } = await FetchJSON<APIResult<APIManga>>(new Request(new URL(`manga/${mangaSlug}`, this.apiUrl)));
+        return new Manga(this, provider, slug_url, rus_name);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let page = 1, run = true; run; page++) {
+            const url = new URL(`manga?page=${page}&site_id[]=${this.siteId}`, this.apiUrl);
+            const { data, meta: { has_next_page } } = await this.mangasTaskPool.Add(() => FetchJSON<APIResult<APIManga[]>>(new Request(url)), Priority.Low);
+            mangaList.push(...data.map(manga => new Manga(this, provider, manga.slug_url, manga.rus_name.trim())));
+            run = has_next_page;
+        }
+        return mangaList;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const { data } = await FetchJSON<APIResult<APIChapter[]>>(new Request(new URL(`manga/${manga.Identifier}/chapters`, this.apiUrl)));
+        return data.reduce((accumulator: Chapter[], chapter) => {
+            let baseTitle = `Том ${chapter.volume} Глава ${chapter.number}`;
+            baseTitle += chapter.name ? ` - ${chapter.name}` : '';
+            const chapters = chapter.branches.map(branch => {
+                const teamName = chapter.branches_count > 1 ? `[${branch.teams.at(0).name}]` : '';
+                const chapterId = JSON.stringify({
+                    branch_id: branch.branch_id?.toString() ?? '',
+                    number: chapter.number,
+                    volume: chapter.volume
+                });
+                return new Chapter(this, manga, chapterId, [baseTitle, teamName].join(' ').trim());
+            });
+            accumulator.push(...chapters);
+            return accumulator;
+        }, []);
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const { branch_id, number, volume } = JSON.parse(chapter.Identifier) as ChapterID;
+        const url = new URL(`manga/${chapter.Parent.Identifier}/chapter?number=${number}&volume=${volume}`, this.apiUrl);
```

Comment:
The path name and search parameters can already be composed in `FetchChapters` and provided as chapter identifier.

Reply:
Aaah i see what you meant. mb